src/main/java/com/iexec/sms/admin/AdminService.java

```diff
@@ -15,15 +15,92 @@
+
+    public AdminService(@Value("${spring.datasource.url}") String datasourceUrl,
+                        @Value("${spring.datasource.username}") String datasourceUsername,
+                        @Value("${spring.datasource.password}") String datasourcePassword) {
+        this.datasourceUrl = datasourceUrl;
+        this.datasourceUsername = datasourceUsername;
+        this.datasourcePassword = datasourcePassword;
+    }
+
+    /**
+     * Creates a backup of the H2 database and saves it to the specified location.
+     *
+     * @param storageLocation The location where the backup file will be saved, must be an existing directory.
+     * @param backupFileName  The name of the backup file.
+     * @return {@code true} if the backup was successful; {@code false} if any error occurs.
+     */
+    boolean createDatabaseBackupFile(String storageLocation, String backupFileName) {
+        // Check for missing or empty storageLocation parameter
+        if (storageLocation == null || storageLocation.isEmpty()) {
+            log.error("storageLocation must not be empty.");
+            return false;
+        }
+        // Check for missing or empty backupFileName parameter
+        if (backupFileName == null || backupFileName.isEmpty()) {
+            log.error("backupFileName must not be empty.");
+            return false;
+        }
+        // Ensure that storageLocation ends with a slash
+        if (!storageLocation.endsWith("/")) {
+            storageLocation += "/";
+        }
+
+        // Check if storageLocation is an existing directory, we don't want to create it.
+        final File directory = new File(storageLocation);
+        if (!directory.exists() || !directory.isDirectory()) {
+            log.error("storageLocation must be an existing directory.");
+            return false;
+        }
+
+        return databaseDump(storageLocation + backupFileName);
+    }
+
+    /**
+     * @param fullBackupFileName complete fileName (location and filename)
+     * @return {@code true} if the backup was successful; {@code false} if any error occurs.
+     */
+    boolean databaseDump(String fullBackupFileName) {
+
+        if (fullBackupFileName == null || fullBackupFileName.isEmpty()) {
+            log.error("fullBackupFileName must not be empty.");
+            return false;
+        }
+
+        try {
+            log.info("Starting the backup process {}", fullBackupFileName);
+            final long start = System.currentTimeMillis();
+            Script.process(datasourceUrl, datasourceUsername, datasourcePassword, fullBackupFileName, "DROP", "");
+            final long stop = System.currentTimeMillis();
+            final long size = (new File(fullBackupFileName)).length();
+            log.info("New backup created [timestamp {}, duration {} ms, size {}]", dateFormat.format(new Date(start)), stop - start, size);
+        } catch (SQLException e) {
+            log.error("SQL error occurred during backup : " + e.getMessage());
```

Comment:
The following should print the full stacktrace:
```suggestion
            log.error("SQL error occurred during backup", e);
```

Reply:
done in local